flixel/addons/editors/ogmo/FlxOgmo3Loader.hx

```diff
@@ -0,0 +1,478 @@
+	 * @param	EntityLayer				The name of the layer the entities are stored in Ogmo editor. Usually `"entities"` or `"actors"`.
+	 */
+	public function loadEntities(EntityLoadCallback:EntityData->Void, EntityLayer:String = "entities"):Void
+	{
+		for (entity in level.getEntityLayer(EntityLayer).entities)
+			EntityLoadCallback(entity);
+	}
+
+	/**
+	 * Loads every decal in a decal layer into a FlxGroup.
+	 *
+	 * IMPORTANT: All decals must be included in one directory!
+	 *
+	 * @param DecalLayer	The name of the layer the decals are stored in Ogmo editor. Usually `"decals"`.
+	 * @param decalsPath 	The path to the directory in which your decal assets are stored.
+	 */
+	public function loadDecals(DecalLayer:String = 'decals', decalsPath:String):FlxGroup
+	{
+		if (decalsPath.charAt(decalsPath.length - 1) != '/')
+			decalsPath += '/';
+		var g = new FlxGroup();
+		for (decal in level.getDecalLayer(DecalLayer).decals)
+		{
+			var s = new FlxSprite(decal.x, decal.y, decalsPath + decal.texture);
+			s.offset.set(s.width / 2, s.height / 2);
+			if (decal.scaleX != null)
+				s.scale.x = decal.scaleX;
+			if (decal.scaleY != null)
+				s.scale.y = decal.scaleY;
+			if (decal.rotation != null)
+				s.angle = project.anglesRadians ? FlxAngle.asDegrees(decal.rotation) : decal.rotation;
+			g.add(s);
+		}
+		return g;
+	}
+
+	/**
+	 * Parse OGMO Editor level .json text
+	 */
+	static function parseLevelJSON(json:String):LevelData
+	{
+		return cast Json.parse(json);
+	}
+
+	/**
+	 * Parse OGMO Editor Project .ogmo text
+	 */
+	static function parseProjectJSON(json:String):ProjectData
+	{
+		return cast Json.parse(json);
+	}
+
+	/**
+	 * Get Tile Layer data matching a given name
+	 */
+	static function getTileLayer(data:LevelData, name:String):TileLayer
+	{
+		for (layer in data.layers)
+			if (layer.name == name)
+				return cast layer;
+		return null;
+	}
+
+	/**
+	 * Get Grid Layer data matching a given name
+	 */
+	static function getGridLayer(data:LevelData, name:String):GridLayer
+	{
+		for (layer in data.layers)
+			if (layer.name == name)
+				return cast layer;
+		return null;
+	}
+
+	/**
+	 * Get Entity Layer data matching a given name
+	 */
+	static function getEntityLayer(data:LevelData, name:String):EntityLayer
+	{
+		for (layer in data.layers)
+			if (layer.name == name)
+				return cast layer;
+		return null;
+	}
+
+	/**
+	 * Get Decal Layer data matching a given name
+	 */
+	static function getDecalLayer(data:LevelData, name:String):DecalLayer
+	{
+		for (layer in data.layers)
+			if (layer.name == name)
+				return cast layer;
+		return null;
+	}
+
+	/**
+	 * Get matching Tileset data from a given name
+	 */
+	static function getTilesetData(data:ProjectData, name:String):ProjectTilesetData
+	{
+		for (tileset in data.tilesets)
+			if (tileset.label == name)
+				return tileset;
+		return null;
+	}
+}
+
+// Parsed .OGMO Project data
```

Comment:
Might want to turn these into `/** */` comments, otherwise they won't show up in completion or API docs.